src/stateStream.ts

```diff
@@ -0,0 +1,82 @@
+import { markName, defaultErrorSubject } from 'rxbeach/internal';
+import { ActionStream } from 'rxbeach/types/helpers';
+import { tag } from 'rxjs-spy/operators';
+import {
+  shareReplay,
+  startWith,
+  switchMap,
+  catchError,
+  skip,
+  filter,
+  tap,
+} from 'rxjs/operators';
+import { combineReducers, RegisteredReducer } from 'rxbeach/reducer';
+import { Subject, of } from 'rxjs';
+
+/**
+ * Create a state stream for a set of reducers.
+ *
+ * The state stream will always emit its last value upon subscription.
+ * On the first subscription to the state stream, it will emit `defaultState`.
+ *
+ * The state stream is not ref counted, which means that the underlying
+ * producer never resets to the initial value. Subsequent subscriptions to
+ * the stream will always get the previously emitted state.
+ *
+ * @param name A name for debugging purposes
+ * @param action$ The action stream
+ * @param defaultState The initial state of the state stream,
+ *                     which is emitted synchronously upon subscription
+ * @param reducers The reducer entries that are combined with `combineReducers`
+ * @see rxbeach.combineReducers
+ * @returns An stream that emits the reduced state
+ */
+export const stateStream = <State>(
```

Comment:
Not 100% settled on this signature. Since there are 4 args, perhaps it's better to take an object? In the end I did not use an object to stay consistent with the rest of rxbeach and since none of the arguments are optinoal

Reply:
I believe a better signature would be `<State>(name: string, defaultState: State, reducers: RegisteredReducer<State, any>[]) => OperatorFunction<UnknownAction, State>`. This of course means this function is a stream operator.

Having this function be a stream operator is more composable, and will make it much nicer to use it together with `useNamespace`, for instance. It also avoids the problem of too many arguments.

Reply:
Gave this a bit more thought and I agree :) 

My initial "fear" was that making it composable would allow someone to misuse it, but I think we should aim to avoid misuse in other ways. Furthermore, your example of `useNamepace` clearly demonstrates why making it composable as an operator function is important, and I can imagine there might be other uses too.